net/client.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * General-purpose TCP client, to be used with corresponding Net::Server object.
+ * Various sink/source-type objects are used for either sending or receiving
+ * data from the server.
+ */
+
+#pragma once
+
+// C++ includes
+#include <limits>
+#include <string>
+#include <vector>
+
+// OpenCV
+#include <opencv2/opencv.hpp>
+
+// local includes
+#include "node.h"
+#include "socket.h"
+
+namespace GeNNRobotics {
+namespace Net {
+class Client
+  : public Node
+  , Socket
+{
+public:
+    Client(const std::string host, int port = DefaultListenPort);
+    ~Client();
+    Socket *getSocket() const override;
+};
+
+/*
+ * Create client, connect to host over TCP
+ */
+Client::Client(const std::string host, int port)
+{
+    // Needed for Windows
+    WSAStartup();
```

Comment:
Yeah, you're right. I stuck `WSAStartup()`/`WSACleanup()` in `Server`/`Client` ctors and dtors because they are only used once per process atm, but this is a lazy assumption.